### PR TITLE
Introduces `author_hasKey` and `author_hasSessionKeys` rpc endpoints

### PR DESCRIPTION
**Pull request by bkchr**
Both endpoints can be used to check if a key is present in the keystore.

- `hasKey` works on with an individual public key and key type. It
checks if a private key for the given combination exists in the
keystore.
- `hasSessionKeys` works with the full encoded session key blob stored
on-chain in `nextKeys`. This requires that the given blob can be decoded
by the runtime. It will return `true`, iff all public keys of the
session key exist in the storage.

Fixes: https://github.com/paritytech/substrate/issues/4696

### CODE REVIEW

File: client/rpc-api/src/author/error.rs [hunk 1]
```diff
@@ -54,6 +54,9 @@ pub enum Error {
 	/// Some random issue with the key store. Shouldn't happen.
 	#[display(fmt="The key store is unavailable")]
 	KeyStoreUnavailable,
+	/// Invalid session keys encoding.
+	#[display(fmt="The session keys are not encoded correctly")]
```
Comment (NikVolf) [inline client/rpc-api/src/author/error.rs]:
```suggestion
	#[display(fmt="Session keys encoded incorrectly")]
```

File: client/rpc/src/author/mod.rs [hunk 2]
```diff
@@ -112,6 +112,23 @@ where
 		).map(Into::into).map_err(|e| Error::Client(Box::new(e)))
 	}
 
+	fn has_session_keys(&self, session_keys: Bytes) -> Result<bool> {
+		let best_block_hash = self.client.chain_info().best_hash;
+		let keys = self.client.runtime_api().decode_session_keys(
+			&generic::BlockId::Hash(best_block_hash),
+			session_keys.to_vec(),
+		)
+		.map_err(|e| Error::Client(Box::new(e)))?
```
Comment (NikVolf) [inline client/rpc/src/author/mod.rs]: no identation

**Comment by NikVolf** [comment]
This really can save some time bootstrapping node/network

File: client/keystore/src/lib.rs [hunk 3]
```diff
@@ -97,25 +98,22 @@ impl Store {
...
 	}
 
 	/// Insert the given public/private key pair with the given key type.
 	///
 	/// Does not place it into the file system store.
-	fn insert_ephemeral_pair<Pair: PairT>(&mut self, pair: &Pair, key_type: KeyTypeId) {
+	fn insert_ephemeral_pair<Pair: PairT>(&mut self, pair: &Pair, seed: &str, key_type: KeyTypeId) {
 		let key = (key_type, pair.public().to_raw_vec());
-		self.additional.insert(key, pair.to_raw_vec());
+		self.additional.insert(key, seed.into());
```
Comment (tomusdrw) [inline client/keystore/src/lib.rs]: The docs for `additional` say it should be `raw public key`, but here it seems we just put `seed` in.

Also we can just put bulshit, cause we don't check that the `pair` is derived from that `seed`, do we?

Comment (bkchr) [reply client/keystore/src/lib.rs]: The docs say: `(KeyTypeId, Raw public key)` -> `Key phrase/seed`
                                      Key                            ->       Value

(the key is build above in exactly this way.

**Pull request merged**